### PR TITLE
make .Summary a proper weatherForecast device

### PR DESCRIPTION
**Pull request by Garfonso**
I tried to add room & function to the .Summary folder. This did not allow type-detector to detect a weather forecast. 

After a bit of work, it detected it. But there still were two things off:
- icon did not contain the icon URL but a number (name?) instead. So I switched the roles and now the icon role points to the iron url as expected
- days were one of, the _d1 Objects are for current day (i.e. forecast.0 and a bit redundant), _d2 for tomorrow (i.e. forecast.1) and so on. I changed that, now it is as expected and in line with the dasWetter and weatherunderground.

### CODE REVIEW

**Comment by algar42** [comment]
Thanks for helping with this! Just to check - did you change the original content of objects tree? In other words, for this update do we need to completely remove previous version and re-install?

**Comment by algar42** [comment]
Just tried to install from your repo. I see that you removed 'forecast.0' from roles completely, but as per type-detector the 'forecast.0' is the required part of the device to detect the weather device:
image
How this works for you? :)

**Comment by Garfonso** [comment]
> 
> 
> Thanks for helping with this! Just to check - did you change the original content of objects tree? In other words, for this update do we need to completely remove previous version and re-install?

No, it does not overwrite the existing objects. Not sure if it should do that. I did just delete the complete "Summary" tree and restarted the adapter. 




> 
> 
> Just tried to install from your repo. I see that you removed 'forecast.0' from roles completely, but as per type-detector the 'forecast.0' is the required part of the device to detect the weather device:
> image
> How this works for you? :)

I don't really know what you did, but in [SummaryObject.json](https://github.com/Garfonso/ioBroker.accuweather/blob/master/lib/summaryObject.json) I added forecast.0 to most of the `Current` objects and to all for d1 (because they also represent the current day, right? So some objects are redundant - at least in my data I see that). 

Also in your screenshot the first part is a regular expression that will allow `wheater.icon` and `wheater.icon.forecast.0`. So in fact `forecast.0` is optional. The main reason Summary was not accepted as weatherForecast device is, that TempMin and TempMax are only there for d1-d5 and not for current and had the roles of forecast.1-forecast.5 which means that TEMP_MAX and TEMP_MIN were missing but required (as you see in your screenshot). 
So basically, what I did, was to "move" all forecasts one day earlier (i.e. forecast.0 = d1 = current day, forecast.1 = d2 = tomorrow and so on). Also the weather.icon.forecast.x expects and icon URL and I did not see that in the WeatherIcon States).

For me it is working good, I now see the weatherForecast object in devices:
accuweather-forecast-device

**Comment by algar42** [comment]
see now. seem to be working fine.
